api/src/modules/widgets/widgets.controller.ts

```diff
@@ -0,0 +1,29 @@
+import { Controller, HttpStatus, Param } from '@nestjs/common';
+import { WidgetsService } from './widgets.service';
+import { TsRestHandler, tsRestHandler } from '@ts-rest/nest';
+import { widgetsContract as c } from '@shared/contracts/base-widgets.contract';
+import { ControllerResponse } from '@api/types/controller.type';
+import { Public } from '@api/decorators/is-public.decorator';
+
+@Controller()
+export class WidgetsController {
+  constructor(private widgetsService: WidgetsService) {}
+
+  @Public()
+  @TsRestHandler(c.getWidgets)
+  async getWidgets(): Promise<ControllerResponse> {
+    return tsRestHandler(c.getWidgets, async ({ query }) => {
+      const widgets = await this.widgetsService.findAllPaginated(query);
+      return { body: widgets, status: HttpStatus.OK };
+    });
+  }
+
+  @Public()
+  @TsRestHandler(c.getWidget)
+  async getWidget(@Param('id') id: string): Promise<any> {
+    return tsRestHandler(c.getWidget, async ({ query }) => {
+      const widget = await this.widgetsService.getById(id, query);
```

Comment:
Nice catch :)

Reply:
@alepefe-vizz I have updated the param to be used the schema validated param, nice one
Regarding this comment:

`Optional: BaseWidget->Id is an integer and we can use z.coerce.number() on the contract in order to coerce the variable type.`

You are right, however the service method `.getById()` expects a string (uuid) as resource identifier. I am not sure if we should be more verbose parsing it in the route handler, but it will get parsed nonetheless in the data access layer. 

Let me know what do you think